src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseTaskDataObject.cs

```diff
@@ -1210,5 +1241,47 @@ internal bool DbFilesLocationAreValid()
             }
             return true;
         }
+
+        /// <summary>
+        /// Cancels the restore operations
+        /// </summary>
+        public override void Cancel()
+        {
+            foreach (Restore restore in RestorePlan.RestoreOperations)
+            {
+                restore.Abort();
```

Comment:
CancelTaskAsync does that after calling this method

---
In reply to: [134043367](https://github.com/Microsoft/sqltoolsservice/pull/438#discussion_r134043367) [](ancestors = 134043367)